cmd/libsnap-confine-private/utils.c

```diff
@@ -310,3 +310,56 @@ int sc_ensure_mkdirat(int fd, const char *name, mode_t mode, uid_t uid, uid_t gi
 int sc_ensure_mkdir(const char *path, mode_t mode, uid_t uid, uid_t gid) {
     return sc_ensure_mkdirat(AT_FDCWD, path, mode, uid, gid);
 }
+
+bool sc_is_path_clean(const char *path) {
```

Comment:
Can you do a sc_is_path_clean_len with an explicit length and a small wrapper that calls it with strlen? This would allow us to validate non-NUL terminated slices too.

Reply:
Separately, how about something as silly as this (pseudocode).

```c
bool sc_is_path_clean(const char *path) {
  return (path != NULL && !(
          contains(path, "/../") || contains(path, "/./")
          || startswith("../") || startswith(path, "./")
          || endswith(path, "/.") || endswith(path, "/..")
          || equals(path, ".") || equals(path, "..")));
}
```

Reply:
@bboozzoo because snap-confine is not about efficiency. It's about having predictable, simple code that is easy to reason about.